src/collection.rs

```diff
@@ -883,6 +896,23 @@ impl<T> IntoIterator for CollectionWithId<T> {
     }
 }
 
+impl<T> std::iter::FromIterator<T> for CollectionWithId<T>
+where
+    T: Id<T>,
+{
+    #![allow(unused_must_use)]
+    fn from_iter<I>(iter: I) -> Self
+    where
+        I: IntoIterator<Item = T>,
+    {
+        iter.into_iter()
+            .fold(CollectionWithId::default(), |mut accumulator, object| {
+                accumulator.push(object);
```

Comment:
There is one problem here. The function `.push()` returns a `Result<>` which means, this can fails.  The current implementation here will just ignore the error totally (usually, there is compiler warning for an unused `Result` but it doesn't show anymore because of `#![allow(unused_must_use)]` above).

This means that the implementation will ignore duplicate (see last test below for an example). The trait `FromIterator` doesn't really provide a way to fail. And I couldn't find a `TryFromIterator` trait and a corresponding `.try_collect()` method (which I guess would have solved my problem here).